uktrade/data-hub-frontend
# Fix the behavior of FieldAddAnother component inside multi-step forms

## Description of change

This PR fixes a bug when the `FieldAddAnother` component resets its state when used inside a step of a form and when the user moved away from (forward or backward) and back to the step where the component was used.

## Test instructions

1. Launch storybook
2. Go to the [Form Elements > FieldAddAnother > Multi Step](http://localhost:65419/?path=/story/form-form-elements-fieldaddanother--multi-step) storybook example
3. Click the _Continue_ button to get to _Step 2_ of the form
4. Click the _Add another thing_ button three times to add three new input fields
5. Fill out some values in the new fields
6. Remove some of the _things_ by clicking on the corresponding _Remove_ button
7. The removed items should disappear
8. Click the _Back_ button on the bottom of the form to go to the previous form step
9. Click the _Continue_ button to get to _Step 2_ of the form
10. Verify that none of the values of the inputs managed by the `FialdAddAnother` component (the _Thing_ inputs) have changed
11. Go to _Step 3_ of the form and back to _Step 2_
12. Verify that none of the values of the inputs managed by the `FialdAddAnother` component (the _Thing_ inputs) have changed

## Screenshots

https://github.com/uktrade/data-hub-frontend/assets/2333157/6ba71d5e-1253-40a5-92be-95fd6a0d2cfc

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
